### PR TITLE
Adds directive to skip query plan cache from comments

### PR DESCRIPTION
**Pull request by rafael**
### Description

Adds directive to skip query plan cache using comments.

### CODE REVIEW

**Comment by rafael** [comment]
I still need to do it for the tablet side.

File: go/vt/vtgate/executor.go [hunk 1]
```diff
@@ -1168,21 +1168,22 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
...
+		return nil, err
+	}
 	if !e.normalize {
-		plan, err := planbuilder.Build(sql, vcursor)
+		plan, err := planbuilder.BuildFromStmt(sql, stmt, vcursor)
 		if err != nil {
 			return nil, err
 		}
-		if !skipQueryPlanCache {
+		if !skipQueryPlanCache && !skipQueryPlanCacheDirective(stmt) {
```
[the remark this answers is not in the document]
Comment (rafael) [reply go/vt/vtgate/executor.go]: That makes sense to me. I'm in favor of only keeping one way.

The only thing that is more of an implementation detail, is how do I get on the tablet that side that we should skip the plan. First thought is to get it the same way I'm doing in the gate (i.e from the query).

File: go/vt/sqlparser/comments.go [hunk 2]
```diff
@@ -252,3 +260,32 @@ func (d CommentDirectives) IsSet(key string) bool {
 	}
 	return false
 }
+
+// SkipQueryPlanCacheDirective returns true if skip query plan cache directive is set to true in query.
+func SkipQueryPlanCacheDirective(stmt Statement) (skipQuerPlanCacheDirective bool) {
+	switch stmt := stmt.(type) {
+	case *Select:
```
Comment (demmer) [inline go/vt/sqlparser/comments.go]: Could you not DRY this by doing:

```
case *Select, *Insert, *Update, *Delete:
   ...
```

I'm not sure if that works for type switches though.

File: go/vt/sqlparser/comments.go [hunk 3]
```diff
@@ -252,3 +260,32 @@ func (d CommentDirectives) IsSet(key string) bool {
 	}
 	return false
 }
+
+// SkipQueryPlanCacheDirective returns true if skip query plan cache directive is set to true in query.
+func SkipQueryPlanCacheDirective(stmt Statement) (skipQuerPlanCacheDirective bool) {
```
Comment (demmer) [inline go/vt/sqlparser/comments.go]: `skipQuerPlanCacheDirective` has a typo :)

Also IMO the named return var is really long here. I'd shorten to `skip` or just omit it altogether and have a local var which defaults to false.

**Pull request merged**